src/index.ts

```diff
@@ -40,10 +40,11 @@ export const isAuthenticatedSession = (session: MaybeSession): session is Authen
 type AuthInput = Pick<AuthConfig, 'scopes'> & {
   msalClient: ClientApplication
   authReplyRoute: string
+  authorizationUrlRequestOverride?: (req: Request) => Partial<AuthorizationUrlRequest>
```

Comment:
Do we want to support async resolution? You might need to go to a third-party service initially, then cache it for subsequent requests.

Reply:
I think it's a decent idea but then this file would get async-virused and it would be a breaking change

Reply:
I agree with @mderriey - I added a commit to make this sync or async callback using the `await Promise.resolve(syncOrAsync)` style

[make the authorizationUrlRequestOverride callback sync or async](https://github.com/MakerXStudio/express-msal/pull/7/commits/a4f957654b15be537f7fdbd70aaf260398aa8742)